tidy3d/web/webapi.py

```diff
@@ -99,16 +99,47 @@ def upload(
     To start the simulation running, must call :meth:`start` after uploaded.
     """
 
-    task_id = _upload_task(
-        simulation=simulation,
-        task_name=task_name,
-        folder_name=folder_name,
-        callback_url=callback_url,
-    )
+    simulation.validate_pre_upload()
 
+    json_string = simulation._json_string()  # pylint:disable=protected-access
+    data = {
+        "taskName": task_name,
+        "callbackUrl": callback_url,
+    }
+    folder = _query_or_create_folder(folder_name)
```

Comment:
I remember seeing some errors related to this function call.  do we want to wrap it in a try: except:?  For example if the folder doesn't exist?

Reply:
May be good to know what errors out exactly? Also, what do we do if it errors - seems like the upload can't continue in that case?

Reply:
yes to both, I wonder if this logic should just go in the `query_or_create_folder` function